### DISCUSSION TITLE
Upvote Issues

### DISCUSSION THREAD

**Discussion by benhatsor** [community/community#5467] [Projects and Issues]
I think it would be really helpful if users could upvote issues in repositories.  
This would help the contributors fix bugs and implement features that matter because they know what the community thinks is important.

> *Note:* GitHub convention is to upvote issues using emotes. However, this functionality is not easy-to-use and is pretty niche. ([You need to click on the little emoji, then click 👍 ](https://github.com/github/feedback/discussions/4921)). What if the issues in a repo had a prominent button? What if they were sorted by upvotes, by default? This is a rather different outlook.

The new [discussions feature](https://docs.github.com/en/discussions) does not currently answer the need, because you cannot [close a Pull Request](https://github.blog/2013-05-14-closing-issues-via-pull-requests/) with a discussion, add a discussion [to a project](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/adding-issues-and-pull-requests-to-a-project-board) or mention a discussion [in a release](https://github.blog/2011-10-12-introducing-issue-mentions/) the same way you can with Issues.

Implementation can be similar to prominent community forums like [Reddit](https://www.reddit.com/) or [Stack Overflow](https://stackoverflow.com/), where this functionality is implemented.

<img width="601" alt="stackoverflow-upvote" src="https://user-images.githubusercontent.com/34835685/132085079-519a6677-3cf0-4391-a0d1-f6b13f960a21.png">

### Related Unanswered Discussions:
- https://github.com/github/feedback/discussions/3097
- https://github.com/github/feedback/discussions/2971
- https://github.com/github/feedback/discussions/3474
- https://github.com/github/feedback/discussions/2948
- https://github.com/github/feedback/discussions/3328

*Please click on the Upvote (🔼 button) and not the +1 emoji so the Github contributors know this is important.*

**Comment by rodber** [comment]
I think that it should be restricted to collaborators. I'm worried about the feature creep this may drive as I've experienced how much some individuals could push hundreds of votes just because they want stuff on their own terms.

**Reply by noraa-junker** [reply]
I would like if the repository owner can set this

**Reply by capaj** [reply]
I would prefer if they can't set this. The end result will be that most project will have it set up for general public. Me as a good open source citizen, I will always upvote issues I encounter. If this is configurable to only collaborators, I will try to search for the upvote button not to find it, or to find it disabled for a small portion of OSS projects.
What's the problem with having an issue with hundreds of votes anyway? It's your repository. No one says you need to do anything, even if it gets hundreds of upvotes.

**Reply by rodber** [reply]
Reason for only contributors is because there's already reactions. People can emoji all what they want, but bring things to the table should be for collaborators only as they will do the job.

**Reply by grabofus** [reply]
> Reason for only contributors is because there's already reactions. People can emoji all what they want, [...]

I think the problem with that is, you can't see the emojis until you click on an issue. This feature could allow you to see engagement on the issues page without having to go through all your issues 1-by-1.

That said.. Would it be possible to allow everyone to upvote an issue, but be able to filter it later on to collaborators or all?

**Reply by skestle** [reply]
This.
> allow everyone to upvote an issue, but be able to filter it later on to collaborators or all

**Comment by TechStudent10** [comment]
Agreed ✔

**Comment by rangav** [comment]
This feature will be very useful to prioritise issues or features.

**Reply by ghost** [reply]
+1

**Comment by matkoniecz** [comment]
> What if they were sorted by upvotes, by default?

That is a quite bad idea, in typical use I prefer to see newest issues first.

**Comment by oerix** [comment]
Upvote from me. We have it for Discussions, why not for Issues? It helps prioritise issues and can be used as another option to sort, which makes it easier to find.

**Comment by rubyFeedback** [comment]
Recently I found some repositories where I do not appear to be able to upvote on something. For instance, github ladybird.

I used to be able to upvote just fine, like 2 weeks ago.

I also don't get any message or explanation about this.

Right now I do not know if this is a bug or a feature. I click on e. g. the thumbs up, it shows up briefly,
then suddenly vanishes. I did not have this behaviour a few weeks ago, so I suspect there is some toggle
option changed at github ladybird.

Either way, if this is not a bug, could the feature be explained to the user? Right now I only see that my
upvote vanishes with a half second delay and I get zero explanations. It would be nice if some explanation
could show up, for instance a slight pop-in widget that vanishes after 5 seconds and has a 'x' to close it
too, that says "this repository does not allow upvotes" or something like that. Right now it is just mega-confusing
to me.
